x-pack/platform/plugins/private/transform/server/lib/alerting/transform_health_rule_type/register_transform_health_rule_type.ts

```diff
@@ -177,12 +184,24 @@ export function getTransformHealthRuleType(
         fieldFormatsRegistry,
       });
 
-      const executionResult = await transformHealthService.getHealthChecksResults(params);
+      const executionResult = await transformHealthService.getHealthChecksResults(
+        params,
+        previousState
+      );
 
       const unhealthyTests = executionResult.filter(({ isHealthy }) => !isHealthy);
 
+      const state: TransformHealthAlertState = {};
+
       if (unhealthyTests.length > 0) {
         unhealthyTests.forEach(({ name: alertInstanceName, context }) => {
+          switch (alertInstanceName) {
+            case TRANSFORM_HEALTH_CHECK_NAMES.notStarted.name:
+              state.notStarted = context.results.map((r) => r.transform_id);
+            case TRANSFORM_HEALTH_CHECK_NAMES.healthCheck.name:
+              state.unhealthy = context.results.map((r) => r.transform_id);
+          }
```

Comment:
I wonder if this will assign correctly? This will overwrite `notStarted/unhealthy` on every `unhealthyTests.forEach` iteration. Should we check if the attributes are populated and push/append if that's the case?

Reply:
It is assigned correctly, as alert instances are created per check type